historical_system_profiles/archiver.py

```diff
@@ -143,9 +146,8 @@ def _check_and_send_notifications(
                 # Add each baseline to our event, then send notification only once
                 # containing the whole list of events for this system
                 drift_found = True
-                # currently using baseline_id in place of baseline_name for testing
-                # will need to make another call to api to get this later
-                event.add_drifted_baseline(baseline_id, baseline_id, comparison)
+                baseline_name = fetch_baselines([baseline_id], service_auth_key, logger)[0]
```

Comment:
I am not sure about this.

IIUC `fetch_baselines` will return a `list` of `dicts` (baselines) according to https://github.com/RedHatInsights/kerlescan/blob/master/kerlescan/service_interface.py#L62 b/c the response is `dict` (`DataPage` - https://github.com/RedHatInsights/system-baseline-backend/blob/master/system_baseline/openapi/api.spec.yaml#L321)

That `dict` had `data` property (`list` - https://github.com/RedHatInsights/system-baseline-backend/blob/master/system_baseline/openapi/api.spec.yaml#L329).

In that `list` we have `dict`s (Baseline - https://github.com/RedHatInsights/system-baseline-backend/blob/master/system_baseline/openapi/api.spec.yaml#L379) - this is the result of `fetch_data` call and thus also the result of `fetch_baselines` call.

Which means that `baseline_name` variable will have first (and only) baseline `dict` stored in it.

So I think we miss `['display_name']` or `.get('display_name')`(with sensible default - `baseline_id` perhaps?) b/c of https://github.com/RedHatInsights/system-baseline-backend/blob/master/system_baseline/openapi/api.spec.yaml#L384


I maybe mistaken as I haven't seen that parts of the code for some time ;-)

Reply:
Definitely forgot [display_name] on Friday, good catch.  I don't think we need to treat it as a dict though because fetch_data can be that or a list, either one, and the baseline api call yields a list so I would expect it to be a list.  Updated this to add [display_name].  Thanks!